### PR TITLE
make pubsub use gapic client

### PR DESCRIPTION
**Pull request by pongad**
No expected change in behavior. This change is to align pubsub with other clients that also use generated gapics. In the future, it'll get updates on retries etc automatically.

### CODE REVIEW

File: google-cloud-clients/pom.xml [hunk 1]
```diff
@@ -660,7 +660,7 @@
           <dependency>
             <groupId>com.puppycrawl.tools</groupId>
             <artifactId>checkstyle</artifactId>
-            <version>6.15</version>
+            <version>6.16</version>
```
Comment (pongad) [inline google-cloud-clients/pom.xml]: This doesn't really belong here, but needed to make this build on my machine at home. See #3577

**Comment by chingor13** [comment]
@pongad Can you resolve the conflicts here?

File: google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java [hunk 2]
```diff
@@ -110,11 +108,17 @@
...
+
+  private SubscriberStub subStub;
+  private final SubscriberStubSettings subStubSettings;
   private final FlowController flowController;
-  private final TransportChannelProvider channelProvider;
-  private final CredentialsProvider credentialsProvider;
-  private final List<Channel> channels;
+  private final int numPullers;
+
+  // private final int numChannels;
```
Comment (chingor13) [inline google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java]: nit: remove these commented out fields

File: google-cloud-clients/google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/SubscriberTest.java [hunk 3]
```diff
@@ -114,6 +114,8 @@ public void testFailedChannel_recoverableError_channelReopened() throws Exceptio
 
   @Test(expected = IllegalStateException.class)
   public void testFailedChannel_fatalError_subscriberFails() throws Exception {
+    System.err.println("testFailedChannel_fatalError_subscriberFails");
```
Comment (chingor13) [inline google-cloud-clients/google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/SubscriberTest.java]: nit: this debug output isn't needed.

File: google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java [hunk 4]
```diff
@@ -62,9 +64,9 @@
   private static final Duration INITIAL_CHANNEL_RECONNECT_BACKOFF = Duration.ofMillis(100);
   private static final Duration MAX_CHANNEL_RECONNECT_BACKOFF = Duration.ofSeconds(10);
   private static final int MAX_PER_REQUEST_CHANGES = 1000;
-  private static final Duration UNARY_TIMEOUT = Duration.ofSeconds(60);
 
   private final SubscriberStub stub;
+  private final int chanAffinity;
```
Comment (chingor13) [inline google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java]: `chanAffinity` -> `channelAffinity`

File: google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java (hunk 2 above)
Comment (pongad) [reply google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java]: done

File: google-cloud-clients/google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/SubscriberTest.java (hunk 3 above)
Comment (pongad) [reply google-cloud-clients/google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/SubscriberTest.java]: whoops done

File: google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java (hunk 4 above)
Comment (pongad) [reply google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java]: done

**Comment by pongad** [comment]
@chingor13 PTAL

**Comment by chingor13** [comment]
Can you add a note in the PR description about what this is affecting? It'll give us more context and improve the release notes.

**Comment by pongad** [comment]
@chingor13 I added the description. I'm not sure if it's worth mentioning in release notes though, this is mostly a clean up with no expected change in behavior.

**Comment by chingor13** [comment]
Integration test failure was a network issue in spanner test.

**Pull request merged**